Examples/Detectors/GeoModelDetector/src/GeoModelDetector.cpp

```diff
@@ -21,4 +19,6 @@ GeoModelDetector::GeoModelDetector(const Config& cfg)
   m_geoModel = Acts::GeoModelReader::readFromDb(m_cfg.path);
 }
 
+GeoModelDetector::~GeoModelDetector() = default;
+
```

Comment:
On my machine I don't see it failing if I revert this change. Since it doesn't hurt, however, we can still make the change.

Reply:
won't this remove the implicitly defaulted move constructor and move assignment? usually I try to follow https://en.cppreference.com/w/cpp/language/rule_of_three

Reply:
In principle, I think this is correct. I'm not sure if the rule of three applies if you explicitly default one of the constructors however.

Sometimes, you need to do this to allow smart pointers to work with forward declarations in headers by pushing the destructor instantiation to a cpp file where the type is complete.

Locally, it didn't seem like this was needed here, but @goetzgaycken reported that he had issues without this change.

Reply:
Ah, actually the doc says that `=default` prevents implicit constructors, you're correct.